include/ppx/application.h

```diff
@@ -515,6 +514,29 @@ class Application
     double            mFirstFrameTime    = 0;
     std::deque<float> mFrameTimesMs;
 
+    // Standard Options
+    struct
+    {
+        // Flags
+        std::shared_ptr<KnobConstant<bool>> pListGpus;
```

Comment:
In general, shared_ptr should be avoided. Is there a reason it's being used here rather than unique_ptr? Or even just a KnobConstant?

Reply:
It should just be a "raw" pointer here with KnobManager keep ownership of the knobs. The usage was from #155 since proper memory management is tricky for people who are new to C++.

Reply:
`shared_ptr` is thread safe and uses extra memory for reference counting, which makes it quite expensive.

The vector should be holding `unique_ptr`, and a reference of the object should be returned.